addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/HeaderBasedSessionManagementMethodType.java

```diff
@@ -0,0 +1,454 @@
+        return new HeaderBasedSessionManagementMethodOptionsPanel();
+    }
+
+    @Override
+    public boolean hasOptionsPanel() {
+        return true;
+    }
+
+    @Override
+    public boolean isTypeForMethod(SessionManagementMethod method) {
+        return method instanceof HeaderBasedSessionManagementMethod;
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        // Nothing to hook
+    }
+
+    private static Map<String, String> listToMap(List<Pair<String, String>> list) {
+        Map<String, String> map = new HashMap<String, String>();
+
+        return map;
+    }
+
+    private static List<Pair<String, String>> mapToList(Map<String, String> map) {
+        List<Pair<String, String>> list = new ArrayList<Pair<String, String>>();
+
+        return list;
+    }
+
+    @Override
+    public SessionManagementMethod loadMethodFromSession(Session session, int contextId)
+            throws DatabaseException {
+        HeaderBasedSessionManagementMethod method = new HeaderBasedSessionManagementMethod();
+
+        Map<String, String> map =
+                EncodingUtils.stringToMap(
+                        session.getContextDataString(
+                                contextId, RecordContext.TYPE_SESSION_MANAGEMENT_FIELD_1, ""));
+
+        method.setHeaderConfigs(mapToList(map));
+
+        return method;
+    }
+
+    @Override
+    public void persistMethodToSession(
+            Session session, int contextId, SessionManagementMethod method)
+            throws UnsupportedSessionManagementMethodException, DatabaseException {
+
+        if (!(method instanceof HeaderBasedSessionManagementMethod)) {
+            throw new UnsupportedSessionManagementMethodException(
+                    "Header based session management type only supports: "
+                            + HeaderBasedSessionManagementMethod.class.getName());
+        }
+        HeaderBasedSessionManagementMethod sessionMethod =
+                (HeaderBasedSessionManagementMethod) method;
+
+        session.setContextData(
+                contextId,
+                RecordContext.TYPE_SESSION_MANAGEMENT_FIELD_1,
+                EncodingUtils.mapToString(listToMap(sessionMethod.getHeaderConfigs())));
+    }
+
+    @Override
+    public void exportData(Configuration config, SessionManagementMethod sessionMethod) {
+        if (!(sessionMethod instanceof HeaderBasedSessionManagementMethod)) {
+            throw new UnsupportedSessionManagementMethodException(
+                    "Header based session management type only supports: "
+                            + HeaderBasedSessionManagementMethod.class.getName());
+        }
+        HeaderBasedSessionManagementMethod method =
+                (HeaderBasedSessionManagementMethod) sessionMethod;
+
+        List<String> list = new ArrayList<>();
+        for (Pair<String, String> pair : method.headerConfigs) {
+            list.add(pair.first + ":" + pair.second);
```

Comment:
The `pair.first` needs to be escaped (or encoded) otherwise the split will not work properly, e.g. if you have a token it will be split in "half" when imported.

Would be good to have tests for the export/import.